packages/sdk/src/helpers/ethers.ts

```
@@ -106,6 +106,23 @@ export function isPolygon(chainId: number | bigint): boolean {
   return chainIdNumber === 137 || chainIdNumber === 80002;
 }
 
+/**
+ * Check if a chain can use the `eth_getMaxFeePerGas`,
+ * `eth_getMaxPriorityFeePerGas`, or `eth_estimateGas`, which are used by the
+ * built-in ethers `getFeeData` method.
+ * @param chainId The chainId of the network as a number or bigint.
+ * @returns A boolean that indicates if `getFeeData` can be used for the chain.
+ */
+export function chainCanGetFeeData(chainId: number | bigint): boolean {
+  const chainIdNumber = typeof chainId === "bigint" ? Number(chainId) : chainId;
+  // Hardhat, Optimism Sepolia, & Filecoin Calibration will return an RPC error
+  return !(
+    chainIdNumber === 31337 ||
```

@dtb I'm wondering if we know for sure that chains with id `31337` always don't have these methods?

---

For background, there was a user in discord using that chainId for a private chain and they were using that chain id.  Also there are some other local dev chains that use this id, e.g. https://chainlist.org/chain/31337

---

hm, yeah that's a good point; we can remove 31337. the main thing this method does is silence an error for `The method "eth_maxPriorityFeePerGas" does not exist / is not available.', data: {…}}`, which will get logged in the console of the Studio/web apps on those 3 chains. but, who cares if it gets logged with LT tbh.

some additional context: `anvil` and `hardhat` don't support the `eth_maxPriorityFeePerGas` method but do support `eth_gasPrice`. e.g., when you try to call the native ethers `getFeeData` on hardhat, the `eth_maxPriorityFeePerGas` method logs an error, but the tx still gets executed. aka we don't _need_ to include 31337 because it'll still work fine.

---

also, we _could_ adjust [this method](https://github.com/tablelandnetwork/tableland-js/blob/2c3b92be798309464b34fe8b37c58fccf2570c7e/packages/sdk/src/helpers/ethers.ts#L93) to destructure the `gasPrice` too. we just have to handle logic to avoid an error for `Cannot send both gasPrice and maxFeePerGas params`. e.g.:
```js
// also get the gas price and conditionally pass it
const { gasPrice, maxFeePerGas, maxPriorityFeePerGas } = await getFeeData(
  signer
);
if (maxFeePerGas != null && maxPriorityFeePerGas != null) {
  opts.maxFeePerGas = maxFeePerGas;
  opts.maxPriorityFeePerGas = maxPriorityFeePerGas;
} else if (gasPrice != null) {
  opts.gasPrice = gasPrice;
}
```
to get the gas price, we'd have to call `eth_gasPrice` directly to avoid the logging error from above.

---

sticking to the removal of 31337 sounds good to me.  If it's annoying in LT we can stop logging those messages in LT? (If that makes sense, we do it in another PR)

---

works for me!